### DISCUSSION TITLE
Custom alerts - Signal API

### DISCUSSION THREAD

**Discussion by kmath313** [TwiN/gatus#882] [Q&A]
Hi All, 
I am using BBhernard's Signal API for a custom alert in Gatus. I have configured it to send the alerts both on error and resolution without any issues. 

The issue I am having though is that the placeholder [RESULT_ERRORS] is not expanding out to list what conditions check failed. Yet when I use one of the pre-configured alert providers I get the list of conditions checks.

I looked at the code for the custom alert and is uses result.Errors but the pre-configured alerts use results.ConditionsResults. 

Firstly - does anyone else have the issue where a custom provider isn't expanding to the failed conditions checks? And secondly is there any reason why the custom provider uses a different error slice to the pre-configured providers?

**Comment by aalmenar** [comment]
Im having some issues with the custom alerting provider since it does not expand results.ConditionResults but results.Errors. This makes the beahaviour of the custom alert totally different from all the other providers. I believe they should be expanded to  the same as other provider or have the custom provider to have an option to enable that (Emojis included) like in the pushover or telegram providers. https://github.com/TwiN/gatus/blob/master/alerting/provider/pushover/pushover.go#L151C105-L167

**Comment by aalmenar** [comment]
I have added a pull request to have RESULT_CONDITIONS placeholder so it can be used with custom alerting module.

https://github.com/TwiN/gatus/pull/1086

**Comment by TwiN** [comment]
Closing this issue, because I would much rather adding support for Signal as an alerting provider than heavily relying on the custom alerting provider.

The purpose of non-custom alerting providers _is_ to custom-build them for the specific provider. The custom alerting provider, on the other hand, is more of a catch-all/backup solution.

@kmath313 if you don't mind, could you provide the custom configuration you're using?

**Reply by kmath313** [reply]
Hi TwiN,
This was the config that I was using.
`alerting:
  default-alert: &defaultAlert
    enabled: tru
    description: health-check failed
    send-on-resolved: true
    failure-threshold: 1
    success-threshold: 1
  discord:
    webhook-url: ${DISCORD_URL}
    <<: *defaultAlert
  custom:
    # url: ${DISCORD_URL} 
    url: "http://REDACTED:8080/v2/send" 
    method: "POST" 
    headers: 
      Content-Type: application/json 
    body: $SIGNAL_MSG 
    <<: *defaultAlert `

$SIGNAL_MSG = `'{"message": "[ALERT_TRIGGERED_OR_RESOLVED]: [ENDPOINT_GROUP] - [ENDPOINT_NAME] - [ALERT_DESCRIPTION] - [RESULT_ERRORS]", "number": "+xxxxxx" , "recipients": [ "+xxxxx" ]}'`

**Reply by TwiN** [reply]
Thank you @kmath313, much appreciated! This summer I plan on adding support for a lot of alerting providers, Signal will likely be one of them, so this is helpful.

**Reply by kmath313** [reply]
Too easy mate. I'm not using gatus atm, but def can spin it up on my cluster for testing. If I can find a home assistant integration I will move back to it. I love that the config is done in a file which means I can integrate it into my FluxCD for the cluster.

**Reply by TwiN** [reply]
I plan on switching to home assistant for my smart home this summer, so I'll probably add support for it as well. Not quite sure what kind of support I'd add though -- what are you thinking? Is there a way to have HA send a notification if an integration tells it to?

**Reply by TwiN** [reply]
Just created #1087, so if you have any details you want to add, feel free to drop them there as well.

**Reply by kmath313** [reply]
I currently use Uptime Kuma, and it has a HA integration. The integration itself do notifications but you certainly could use a configured HA notification service to do that. I use the integration as part of a dashboard. I have about 10-12 internal services that I monitor and about 6 external. The integration allows me to put what amounts to the Gatus status badges into my HA dashboard. I would imagine that you could probably have HA do a standard HTTP/S request to the Gatus API endpoint that creates the status badges.
